src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -1550,8 +1554,18 @@ public void setWaitAtBeginningFactor(double waitAtBeginningFactor) {
         }
     }
 
-    public void banTrip(FeedScopedId trip) {
-        bannedTrips.put(trip, BannedStopSet.ALL);
+    /**
+     * Bans various tripID sequences as needed from a graphPath. The specific sequence of trip IDs found in the path
+     * won't be used in the graph path finding if the sequence of trips is detected in any order in another path. See
+     * {@link TripTimes#tripOrTripSequenceIsBanned(State, int)} for further details.
+     */
+    public void banTripsInPath(GraphPath path) {
+        List<FeedScopedId> tripIds = path.getTrips();
+        List<FeedScopedId> callAndRideTripIds = path.getCallAndRideTrips();
+        // ban a sequence of trips if none of the tripIds were call and ride trips
+        if (!tripIds.stream().anyMatch(tripId -> callAndRideTripIds.contains(tripId))) {
```

Comment:
Intellij simplified this to
```suggestion
        if (tripIds.stream().noneMatch(callAndRideTripIds::contains)) {
```

Reply:
Refactored in 6adec27faa2e744f94cd91e4ae4d337a5c25c7c7